modules/common/util/template_util.go

```diff
@@ -138,6 +139,57 @@ func ExecuteTemplate(templateFile string, data interface{}) (string, error) {
 	return renderedTemplate, nil
 }
 
+// template functions
+var tmpl *template.Template
+
+// template function which allows to execute a template from within
+// a template file.
+// name - name of the template as defined with with `{{define "some-template"}}your template{{end}}
+// data - data to pass into to render the template for all can use `.`
+func execTempl(name string, data interface{}) (string, error) {
+	buf := &bytes.Buffer{}
+	err := tmpl.ExecuteTemplate(buf, name, data)
+	return buf.String(), err
+}
+
+// template function to indent the template with n tabs
+func indent(n int, in string) (out string) {
+	s := bufio.NewScanner(bytes.NewReader([]byte(in)))
+	for s.Scan() {
+		line := strings.TrimSpace(s.Text())
+		for i := 0; i < n; i++ {
+			line = "\t" + line
+		}
+		out += line + "\n"
+	}
+	return out
+}
+
+// template function to remove empty lines if there are > n continuous empty lines
+func removeNewLines(n int, in string) (out string) {
```

Comment:
IIUTC, this allow people to have n number of lines in between.
I think we may not need this in case conf files, but may be for other templates!